sdk/aqueduct/integrations/mongodb_integration.py

```diff
@@ -0,0 +1,169 @@
+            collection=self._collection_name, query_serialized=serialized_args
+        )
+        param_names = find_parameter_names(serialized_args)
+        param_artifacts = find_parameter_artifacts(self._dag, param_names)
+        for artf in param_artifacts:
+            if artf.type != ArtifactType.STRING:
+                raise InvalidUserArgumentException(
+                    "The parameter `%s` must be defined as a string. Instead, got type %s"
+                    % (artf.name, artf.type)
+                )
+        param_artf_ids = [artf.id for artf in param_artifacts]
+        op_id = generate_uuid()
+        output_artf_id = generate_uuid()
+        apply_deltas_to_dag(
+            self._dag,
+            deltas=[
+                AddOrReplaceOperatorDelta(
+                    op=Operator(
+                        id=op_id,
+                        name=op_name,
+                        description=description,
+                        spec=OperatorSpec(
+                            extract=ExtractSpec(
+                                service=self._metadata.service,
+                                integration_id=self._metadata.id,
+                                parameters=mongo_extract_params,
+                            )
+                        ),
+                        inputs=param_artf_ids,
+                        outputs=[output_artf_id],
+                    ),
+                    output_artifacts=[
+                        ArtifactMetadata(
+                            id=output_artf_id,
+                            name=artifact_name_from_op_name(op_name),
+                            type=ArtifactType.TABLE,
+                        ),
+                    ],
+                ),
+            ],
+        )
+
+        if execution_mode == ExecutionMode.EAGER:
+            # Issue preview request since this is an eager execution.
+            artifact = artifact_utils.preview_artifact(self._dag, output_artf_id)
+            assert isinstance(artifact, TableArtifact)
+            return artifact
+        else:
+            # We are in lazy mode.
+            return TableArtifact(self._dag, output_artf_id)
+
+    def config(self, update_mode: LoadUpdateMode) -> SaveConfig:
+        return SaveConfig(
+            integration_info=self._metadata,
+            parameters=RelationalDBLoadParams(table=self._collection_name, update_mode=update_mode),
+        )
+
+
+class MongoDBIntegration(Integration):
+    """
+    Class for MongoDB integration. This works similar to mongo's `Database` object:
+
+    mongo_integration = client.integration("my_integration_name")
+    my_table_artifact = mongo_integration.collection("my_collection").find({})
+    """
+
+    def __init__(self, dag: DAG, metadata: IntegrationInfo):
+        self._dag = dag
+        self._metadata = metadata
+
+    def collection(self, name: str) -> Any:
+        """
+        `collection` returns a specific collection object to call `.find()` method.
```

Comment:
1) I think starting comments with `<name of function>` is a golang thing. Might be worth pattern matching what's in, for example, `aqueduct_client.py`?

2) Doesn't this return a `MongoDBCollectionIntegration`? I don't think the typing should be `Any`.